packages/context/test/unit/resolver.test.ts

```diff
@@ -85,9 +92,140 @@ describe('constructor injection', () => {
       ) {}
     }
 
-    const t = instantiateClass(TestClass, ctx) as TestClass;
+    const t = instantiate(TestClass, ctx) as TestClass;
     expect(t.fooBar).to.eql('FOO:BAR');
   });
+
+  it('reports circular dependencies of two bindings', () => {
+    const context = new Context();
+
+    // Declare two interfaces so that they can be used for typing
+    interface XInterface {}
+    interface YInterface {}
+
+    class XClass implements XInterface {
+      constructor(@inject('y') public y: YInterface) {}
+    }
+
+    class YClass implements YInterface {
+      constructor(@inject('x') public x: XInterface) {}
+    }
+
+    context.bind('x').toClass(XClass);
+    context.bind('y').toClass(YClass);
+    expect(() => context.getSync('x')).to.throw(/Circular dependency/);
```

Comment:
The error message should include keys of the bindings that are forming a circle, otherwise users will have hard time figuring out where is the problem and how to fix it. Please add a check for that too.